dot/parachain/candidate-validation/worker_test.go

```diff
@@ -0,0 +1,165 @@
+package candidatevalidation
+
+import (
+	"testing"
+	"time"
+
+	parachain "github.com/ChainSafe/gossamer/dot/parachain/runtime"
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func Test_worker_executeRequest(t *testing.T) {
+	candidateReceipt, validationCode := createTestCandidateReceiptAndValidationCodeWParaId(t, 1000)
+
+	validationRuntime, err := parachain.SetupVM(validationCode)
+	require.NoError(t, err)
+
+	ctrl := gomock.NewController(t)
+	t.Cleanup(ctrl.Finish)
+
+	expectedValidationResult := &ValidationResult{
+		ValidResult: &Valid{
+			CandidateCommitments: parachaintypes.CandidateCommitments{
+				HeadData: parachaintypes.HeadData{Data: []byte{2, 0, 0, 0, 0, 0, 0, 0, 123, 207, 206, 8, 219, 227,
+					136, 82, 236, 169, 14, 100, 45, 100, 31, 177, 154, 160, 220, 245, 59, 106, 76, 168, 122, 109,
+					164, 169, 22, 46, 144, 39, 103, 92, 31, 78, 66, 72, 252, 64, 24, 194, 129, 162, 128, 1, 77, 147,
+					200, 229, 189, 242, 111, 198, 236, 139, 16, 143, 19, 245, 113, 233, 138, 210}},
+				ProcessedDownwardMessages: 0,
+				HrmpWatermark:             1,
+			},
+			PersistedValidationData: parachaintypes.PersistedValidationData{
+				ParentHead: parachaintypes.HeadData{Data: []byte{1, 0, 0, 0, 0, 0, 0, 0, 1,
+					2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7,
+					8, 9, 0, 1, 2, 48, 246, 146, 178, 86, 226, 64, 9,
+					188, 179, 77, 14, 232, 77, 167, 60, 41, 138, 250, 204, 9, 36, 224, 17, 5, 226, 235,
+					15, 1, 168, 127, 226}},
+				RelayParentNumber: 1,
+				RelayParentStorageRoot: common.MustHexToHash(
+					"0x50c969706800c0e9c3c4565dc2babb25e4a73d1db0dee1bcf7745535a32e7ca1"),
+				MaxPovSize: 2048,
+			},
+		},
+	}
+
+	mockValidationInstance := NewMockValidatorInstance(ctrl)
+	mockValidationInstance.EXPECT().ValidateBlock(gomock.Any()).DoAndReturn(func(parachain.
+		ValidationParameters) (*parachain.ValidationResult, error) {
+		time.Sleep(3 * time.Second) // sleep to simulate execution time
```

Comment:
does it really need to be this long?

Reply:
This is 3 seconds since the default timeout period is 2 seconds.

Reply:
This test case is for checking the condition when validate block takes longer than expected.  The default timeout period is 2 seconds, which is calculated by the `determineTimeout` function.  Instead of interfacing this function to make the test default timeout shorter I decided instead to mock a long running validate block process.  Thus not having to create an interface just for testing.

Reply:
I see